dev-docs/design/how-lit-works.md

```diff
@@ -0,0 +1,320 @@
+# How lit-html Works
+
+## About this Document
+
+This is a high-level overview for how lit-html works, what makes it fast, and how the code is organized. It's intended for contributors, or anyone who is interested in the inner workings of lit-html.
+
+All the source code described by this document can be found in [`lit-html.ts`](https://github.com/lit/lit/blob/main/packages/lit-html/src/lit-html.ts).
+
+The source code itself is generally optimized to be read and we try to be as clear and self-documenting as possible, erring on the side of verbosity, with numerous comments when needed to explain why code is written as it is. We expect the source to be optimized by tools, so we can keep it understandable by humans.
+
+If anything is unclear or you have questions, please open an issue and we'll do our best to clarify either the source or this document.
+
+# What is lit-html?
+
+lit-html is an HTML templating library. Templates are written in JavaScript by mixing static HTML strings and dynamic JavaScript values using template literals. lit-html enables a functional / UI-as-data programming model, fast initial rendering, and fast updates that minimally update DOM when state changes.
+
+The key feature that enables this is separating static parts of templates from the dynamic parts with template literals, and never traversing or updating the static parts after the initial render.
+
+## Comparison to Virtual DOM
+
+lit-html is comparable in many ways to virtual-DOM approaches, but it works without storing a separate representation of the DOM in memory, or computing DOM diffs like virtual-DOM libraries must do.
+
+- Where VDOMs work at the level of the DOM representation of a UI, lit-html works at the _value_ level.
+- lit-html UIs are values, like VDOM, but instead of each DOM node being represented as a single value, the DOM structure is represented by a reference to a template.
+- lit-html only tracks the tree of dynamic values (called `Parts`) that is more sparse than the associated DOM tree, so traversing dynamic values when necessary is faster.
```

Comment:
Working on some visuals now.

Reply:
Instead of expanding out this part of the document, I've cut it. This sort of comparison could live in a separate document. Instead I referenced Justin's video as well as the JS Frameworks benchmark to provide references to how we value performance and how it's done.